macros/contexts/contextNondecimalBase.pl

```diff
@@ -0,0 +1,268 @@
+
+=head1 NAME
+
+contextNonDecimalBase.pl - Implements a MathObject class and context for numbers
+in non-decimal bases
+
+=head1 DESCRIPTION
+
+This context implements positive integers and some operations on integers in a non-decimal base
+greater than or equal to 2.  The numbers will be stored internally in decimal, though parsed
+and shown in the chosen base.
+
+In addition, basic integer arithemetic (+,-,*,^) are available for these numbers, however because it
+is assumed that all numbers are integers, division '/' is disallowed.
+
+The original purpose for this is simple conversion and operations in another base, however
+it is not limited to this.
+
+To use a non-decimal base MathObject, first load the contextNondecimalBase.pl file:
+
+    loadMacros('contextNondecimalBase.pl');
+
+and then select the appropriate context -- one of the following:
+
+    Context('NondecimalBase')->setBase(5);
+
+
+will now interprets those in Compute and student answers in base 5.
+
+    $a = Compute('104');
+    $b = Compute('233');
+    $sum = Compute("$a+$b"); # this is the base-5 number 342 (decimal 97)
+
+For many problems, one may wish to not allow operators in the student answers.  Use
+'LimitedNondecimalBase' for this.
+
+    Context('LimitedNondecimalBase')->setBase(5);
+
+    $a = Compute('104');
+    $b = Compute('233');
+    $sum = Compute("$a+$b"); # There will be an error on this line now.
+
+In both 'NondecimalBase' and 'LimitedNondecimalBase', another option is to pass the
+digits used for the number to the C<setBase> method.  For example, if one wants to use base-12
+and use the alternative digits 0..9,'T','E', then
+
+    Context('NondecimalBase')->setBase([0..9,'T','E']);
+
+Then one can use the digits 'T' and 'E' in a number like:
+
+    Compute('9TE');
+
+=head2 Sample PG problem
+
+A simple PG problem that asks a student to convert a number into base-5 may include:
+
+    Context('LimitedNondecimalBase')->setBase(5);
+
+    # decimal number picked randomly.
+    $a = random(130,500);
+    $a_5 = Compute(convertBase($a, to => 5));
```

Comment:
Actually, there is an easier way to do this.  Since the context has `Real` set to be a non-decimal real in the current base, using `Real($a)` is enough to get you the desired `$a_5` without converting by hand.  E.g.,

``` perl
$a = random(130, 500);
$a_5 = Real($a);
```

is all that is needed.  Or it could be

``` perl
$a_5 = Real(random(130, 500));
$a = $a->value;
```

if you happen to have the non-decimal real first.  E.g.,

``` perl
$a = Compute("104");
$b = Compute("233");
$sum_5  = $a + $b;      #  = 342
$sum_10 = $sum->value;  #  = 97
```


The upshot is that converting into the context's base can be done using `Real()` and out of it via `->value`.

Reply:
Again, I think I understand now. Thanks.